storage/src/Database.cpp

```diff
@@ -165,6 +175,25 @@ bool Database::store(const std::string& hash, const std::string& pubKey,
     return result;
 }
 
+bool Database::bulk_store(const std::vector<service_node::storage::Item>& items) {
+    char *errmsg = 0;
+    if (sqlite3_exec(db, "BEGIN TRANSACTION;", NULL, NULL, &errmsg) != SQLITE_OK) {
+        return false;
+    }
+
+    for (const auto& item : items) {
+        if (!store(item.hash, item.pub_key, item.data, item.ttl, item.timestamp, item.nonce, IGNORE)) {
+            sqlite3_exec(db, "ROLLBACK;", NULL, NULL, &errmsg);
```

Comment:
I'm not sure if we want to rollback if one item fails. What are your thoughts?

Reply:
Yeah, I guess we should try insert what we can.
Will change.

Reply:
Realised `store` would never return false when setting the IGNORE flag, but can still throw when things go south. So I wrapped the call in a try/catch.